lib/main.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |x|
+      new_arr << x if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all(arg = nil)
+    unless block_given?
+      falsy = 0
+      my_each do |x|
+        falsy += 1 if x == false || x.nil?
+      end
+      return falsy.zero? if arg.nil?
+
+      return self == [arg]
+    end
+
+    pos = 0
+    my_each do |x|
+      pos += 1 if yield(x) == true
+    end
+    pos == size
+  end
+
+  def my_any(arg = nil)
+    #     unless block_given?
+    if arg.nil?
+      count = 0
+      my_each do |x|
+        count += 1 if x != false
+      end
+      return count.positive?
+    end
+    #     end
+
+    unless arg.nil?
+      count = 0
+      my_each do |x|
+        count += 1 if x == arg
+      end
+      return count.positive?
+    end
+
+    my_each do |x|
+      return true if yield(x) == true
+    end
+    false
+  end
+
+  def my_none
+    pos = 0
+    my_each do |x|
+      pos += 1 if yield(x) == false
```

Comment:
what if we try

```
irb(main):051:0> ["hi", "now"].none?(5)
=> true
```

the method should be able to take an argument as well

you can find all the expected behaviors for the enumerable methods [here](https://ruby-doc.org/core-2.5.0/Enumerable.html)